bsdbpf/bsd_bpf_sniffer.go

```diff
@@ -143,19 +147,30 @@ func NewBPFSniffer(iface string, options *Options) (*BPFSniffer, error) {
 		// higher level protocol analyzers will not need this
 		err = syscall.SetBpfHeadercmpl(sniffer.fd, enable)
 		if err != nil {
-			return nil, err
+			goto err
 		}
 	}
 
 	if sniffer.options.Promisc {
 		// forces the interface into promiscuous mode
 		err = syscall.SetBpfPromisc(sniffer.fd, enable)
 		if err != nil {
-			return nil, err
+			goto err
+		}
+	}
+
+	if !sniffer.options.SeeSent {
+		// See sent is set by default, need to turn it off
+		err = unix.IoctlSetInt(sniffer.fd, syscall.BIOCSSEESENT, 0)
```

Comment:
```suggestion
		err = unix.IoctlSetPointerInt(sniffer.fd, syscall.BIOCSSEESENT, 0)
```

Reply:
@Doridian, I do not understand this suggestion. Can you please elaborate on it? Here is a definition from bpf man page saying it is **u_int**:
```
     BIOCSSEESENT
     BIOCGSEESENT    u_int These commands are	obsolete but left for compati-
		     ability.  ... This flag is initialized to
		     one by default.
```
from https://man.freebsd.org/cgi/man.cgi?bpf(4)

Thank you for the review

Reply:
@srgg Most write `ioctl` calls actually expect pointers to the values you set. Including `SEESENT`.
This is based on automated testing of my VPN project, where using only `IoctlSetPointerInt` yielded a functional tunnel (the tunnel server, when seeing its own packets, just freaks out as it then tries to relay its own packets back)

Reply:
@Doridian, I see your point. Curious if it is a difference between Darwin and BSD? And AFAIK, it works on Darwin. 

Agreed IOCTLs are weird, but they should be used as expected by the correspondent kernel module (and these two calls are pretty different, I've provided a code snippet below). I cannot access the project I worked on when this PR was created. Can I merge your suggestion w/o extra verification? 


```
// IoctlSetInt performs an ioctl operation which sets an integer value
// on fd, using the specified request number.
func IoctlSetInt(fd int, req uint, value int) error {
	return ioctl(fd, req, uintptr(value))
}

// IoctlSetPointerInt performs an ioctl operation which sets an
// integer value on fd, using the specified request number. The ioctl
// argument is called with a pointer to the integer value, rather than
// passing the integer value directly.
func IoctlSetPointerInt(fd int, req uint, value int) error {
	v := int32(value)
	return ioctl(fd, req, uintptr(unsafe.Pointer(&v)))
}

```

Reply:
@srgg I was testing this on GitHub's macOS runners (and my MacBook), and only `IoctlSetPointerInt` worked. I actually did not test this on a more "normal" BSD.